src/store/reducers/todos.js

```diff
@@ -1,11 +1,64 @@
 export default ( todos = [], { type, ...payload } ) => {
   switch ( type ) {
+
     case 'ADD_TODO_ITEM':
-      return [ ...todos, payload ];
+
+      let _todo_item = { 
+      	...payload,
+      	complete: false,
+      	mode: 'read',
+      };
+
+      return [ ...todos, _todo_item ];
+
       break;
+
+    case 'TOGGLE_TODO_ITEM':
+
+    	// extracting item from state
+    	// and setting complete atttribute
+    	// depending on previous complete status
+    	let { index } = payload;
+    	_todo_item = Object.assign({}, todos[index], { complete: !todos[index].complete })
```

Comment:
thanks,

did not think about using map, much cleaner

and I definitely should have added ids to the items